src/app/tests/TestWriteInteraction.cpp

```diff
@@ -628,6 +652,52 @@ TEST_F(TestWriteInteraction, TestWriteHandlerInvalidateFabric)
 
 #endif
 
+// This test sends an invalid (because there is no application payload at all)
+// Write Request message and makes sure that a correct Status Response is
+// received.
+TEST_F_FROM_FIXTURE(TestWriteInteraction, TestWriteHandlerReceiveEmptyWriteRequest)
+{
+    Messaging::ReliableMessageMgr * rm = GetExchangeManager().GetReliableMessageMgr();
+    // Shouldn't have anything in the retransmit table when starting the test.
+    EXPECT_EQ(rm->TestGetCountRetransTable(), 0);
+
+    TestWriteClientCallback writeCallback;
+    auto * engine = chip::app::InteractionModelEngine::GetInstance();
+    EXPECT_EQ(engine->Init(&GetExchangeManager(), &GetFabricTable(), app::reporting::GetDefaultReportScheduler()), CHIP_NO_ERROR);
+
+    GetLoopback().mDroppedMessageCount = 0;
+    GetLoopback().mSentMessageCount    = 0;
+    GetLoopback().mNumMessagesToDrop   = 0;
+
+    // Just send an empty message claiming to be a write request.
+    System::PacketBufferHandle emptyMessage = System::PacketBufferHandle::New(Crypto::CHIP_CRYPTO_AEAD_MIC_LENGTH_BYTES);
+
+    TestExchangeDelegate delegate;
+    delegate.mLastStatusParsedSuccessfully = false;
+
+    auto exchange = NewExchangeToAlice(&delegate);
+    EXPECT_NE(exchange, nullptr);
+
+    EXPECT_EQ(exchange->SendMessage(Protocols::InteractionModel::MsgType::WriteRequest, std::move(emptyMessage),
+                                    Messaging::SendMessageFlags::kExpectResponse),
+              CHIP_NO_ERROR);
+
+    DrainAndServiceIO();
+
+    EXPECT_EQ(InteractionModelEngine::GetInstance()->GetNumActiveWriteHandlers(), 0u);
+    EXPECT_EQ(GetLoopback().mSentMessageCount, 3u); // Request, response, ack
+    EXPECT_EQ(GetLoopback().mDroppedMessageCount, 0u);
+
+    EXPECT_TRUE(delegate.mLastStatusParsedSuccessfully);
+    EXPECT_EQ(delegate.mLastStatus, Protocols::InteractionModel::Status::InvalidAction);
+
+    engine->Shutdown();
```

Comment:
NIT: Maybe some kind of followup (for all test cases) should be here to move that part to test `SetUp`/`TearDown`?

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/37084, good idea.